commands/utils/packagehandlers/yarnpackagehandler.go

```diff
@@ -19,5 +33,50 @@ func (yarn *YarnPackageHandler) UpdateDependency(vulnDetails *utils.Vulnerabilit
 }
 
 func (yarn *YarnPackageHandler) updateDirectDependency(vulnDetails *utils.VulnerabilityDetails, extraArgs ...string) (err error) {
-	return yarn.CommonPackageHandler.UpdateDependency(vulnDetails, extraArgs...)
+	executableYarnVersion, err := getYarnVersion()
+	if err != nil {
+		return
+	}
+
+	var tmpDirToDelete string
+	isV2AndAbove := version.NewVersion(executableYarnVersion).Compare(yarnV2Version) <= 0
+
+	if isV2AndAbove {
+		vulnDetails.Technology.SetPackageInstallationCommand(yarnV2PackageUpdateCmd)
+	} else {
+		vulnDetails.Technology.SetPackageInstallationCommand(yarnV1PackageUpdateCmd)
+		tmpDirToDelete, err = fileutils.CreateTempDir()
+		if err != nil {
+			return
+		}
+		extraArgs = append(extraArgs, modulesFolderFlag+tmpDirToDelete)
+	}
+	err = yarn.CommonPackageHandler.UpdateDependency(vulnDetails, extraArgs...)
+	if err != nil {
+		return
+	}
+
+	if !isV2AndAbove {
+		err = fileutils.RemoveTempDir(tmpDirToDelete)
+	}
+	return
+}
+
+// getYarnVersion gets the project's executed yarn version. This is required for fetching the correct command for updating packages
+func getYarnVersion() (executableYarnVersion string, err error) {
+	workingDirectory, err := coreutils.GetWorkingDirectory()
+	if err != nil {
+		err = errors.New("couldn't fetch current working directory: " + err.Error())
+		return
+	}
+	yarnExecutablePath, err := biUtils.GetYarnExecutable()
+	if err != nil {
+		err = errors.New("couldn't fetch yarn executable: " + err.Error())
+		return
+	}
+	executableYarnVersion, err = biUtils.GetVersion(yarnExecutablePath, workingDirectory)
+	if err != nil {
+		err = errors.New("couldn't get yarn executed version: " + err.Error())
```

Comment:
the error returned from biUtils.GetVersion is already a nice error that we don't need to wrap